dronekit_sitl/__init__.py

```diff
@@ -160,7 +206,28 @@ def download(self, system, version, target=None, verbose=False):
 
         return download(system, version, target, verbose=verbose)
 
-    def launch(self, args, verbose=False, await_ready=False, restart=False, wd=None, use_saved_data=False):
+    def emit_sitl_stdout(self, line):
+        sys.stdout.write("SITL-%d> " % (self.instance,))
+        sys.stdout.write(line.decode(sys.getdefaultencoding()) if six.PY3 else line)
+
+    def emit_sitl_stderr(self, line):
+        sys.stdout.write("SITL-%d.stderr> " % (self.instance,))
+        sys.stdout.write(line.decode(sys.getdefaultencoding()) if six.PY3 else line)
+
+    def _sitl_reader(self):
+        while not self.sitl_reader_should_quit:
+            line = self.stdout.readline(timeout=1)
+            if line is not None:
+                self.emit_sitl_stdout(line)
+            line = self.stderr.readline(timeout=1)
+            if line is not None:
+                self.emit_sitl_stderr(line)
+            if not main_thread().is_alive():
```

Comment:
why did you make `main_thread` an global func instead of a member of this class?